packages/edit-widgets/src/components/widget-area/index.js

```diff
@@ -9,13 +16,35 @@ import {
 } from '@wordpress/block-editor';
 import { withDispatch, withSelect } from '@wordpress/data';
 
+const { mediaUpload } = mediaUtils;
+
 function WidgetArea( {
 	blockEditorSettings,
 	blocks,
 	initialOpen,
 	updateBlocks,
 	widgetAreaName,
+	hasUploadPermissions,
 } ) {
+	const settings = useMemo(
+		() => {
+			if ( ! hasUploadPermissions ) {
+				return blockEditorSettings;
+			}
+			const mediaUploadBlockEditor = ( { onError, ...argumentsObject } ) => {
```

Comment:
Having all of this logic in the component's render function is a little distracting. Maybe we could pull this function out into a new file which mirrors `packages/editor/src/utils/media-upload/index.js`?

Reply:
I extracted the settings computation to a specific function. That makes the code easier to understand.